File: gosrc/leason3/dog.go
package leason3

import (
	"fmt"
)

func NewDog(name string) Animal {
	ret := &dog{name}
	return ret
}

type dog struct {
	name string
}

func (d *dog) Eat() {
	fmt.Println("bone")
}

func (d *dog) Name() string {
	return d.name
}

File: src/com/heiwawu/example/leason3/impl/Dog.java
package com.heiwawu.example.leason3.impl;

import com.heiwawu.example.leason3.Animal;

public class Dog implements Animal{
    private String name;
    public Dog(String name){
        this.name = name;
    }
    @Override
    public void eat() {
        System.out.println("bone");
    }

    @Override
    public String name() {
        return this.name;
    }
}
